src/lint.ts

```diff
@@ -301,11 +304,19 @@ export class Linter {
     }
     let root;
     try {
-      root = yaml.safeLoad(this.inYaml);
+      root = yaml.safeLoadAll(this.inYaml);
```

Comment:
I'm not sure I agree with this 100%.

We added some stuff to the CLI to lint multidoc YAML in #163 

I think the doc selection should happen outside the core linter method. I don't think it's harder to do it that way, and reduces responsibilities of `linter.lint()`.

Reply:
I don't think that 'doc selection' is the only reason to move this to `linter.lint()`, though. For one, vendor web calls `linter.lint()` directly and so does not receive the benefits of a wrapper function - and more relevant here, some of the information about the parsed documents may be relevant when parsing an individual document.

Reply:
I don't think "vendor web calls this with multiple docs" is a reason to move additional YAML parsing logic into this method.

What I'm getting at here is: have we considered letting the client be responsible for choosing the doc? It's not a lot to ask of venweb, and this method is already supler huge.

There may be other reasons to do this here, but I don't see a compelling reason to increase the complexity in the library method just to save clients some code (parsing) that isn't the core focus of replicated-lint (linting)

Reply:
You could easily expose the doc selection logic as a helper function, to be called by the client to select a doc before calling `linter.lint()`.

Essentially two narrower interfaces instead of one increasingly broad one.